plugins/include/engine.inc

```diff
@@ -24,242 +24,1271 @@
+ *
+ * @return              Entity index if an entity was found, 0 otherwise
+ */
+native find_ent_by_owner(iIndex, const szClass[], iOwner, iJghgType = 0);
+
+/**
+ * Searches entities in the world, starting at a specified index and matching by
+ * target.
+ *
+ * @param iIndex    Entity index to start from
+ * @param szClass   Target to match
+ *
+ * @return          Entity index if an entity was found, 0 otherwise
+ */
 native find_ent_by_target(iIndex, const szClass[]);
+
+/**
+ * Searches entities in the world, starting at a specified index and matching by
+ * targetname.
+ *
+ * @param iIndex    Entity index to start from
+ * @param szClass   Targetname to match
+ *
+ * @return          Entity index if an entity was found, 0 otherwise
+ */
 native find_ent_by_tname(iIndex, const szClass[]);
+
+/**
+ * Searches entities in the world, starting at a specified index and matching by
+ * classname and model.
+ *
+ * @param iIndex    Entity index to start from
+ * @param szClass   Classname to match
+ * @param szModel   Model to match
+ *
+ * @return          Entity index if an entity was found, 0 otherwise
+ */
 native find_ent_by_model(iIndex, const szClass[], const szModel[]);
+
+/**
+ * Searches for entities inside a sphere, starting at a specified index.
+ *
+ * @param start_from_ent    Entity index to start from
+ * @param origin            Center of sphere
+ * @param radius            Sphere radius
+ *
+ * @return                  Entity index if an entity was found, 0 otherwise
+ */
 native find_ent_in_sphere(start_from_ent, const Float:origin[3], Float:radius);
 
-//this will CBaseEntity::Think() or something from the entity
+/**
+ * Calls the DispatchThink() game DLL function on an entity, triggering it to
+ * think if applicable.
+ *
+ * @note DispatchThink() checks the entity for the FL_DORMANT flag - if it is
+ *       set, the entity will not proceed to think. It will first call the
+ *       class-specific think function and eventually CBaseEntity::Think(), thus
+ *       triggering other think hooks and forwards.
+ *
+ * @param entity    Entity index
+ *
+ * @noreturn
+ * @error           If an invalid entity index is provided, an error will be
+ *                  thrown.
+ */
 native call_think(entity);
 
-/* Is entity valid? */
+/**
+ * Returns if an entity index is valid (as required by other engine natives).
+ *
+ * @note Engine considers an entity index valid if it is in the range between 1
+ *       and the maximum number of entities possible. The index also has to
+ *       point to an existing entity or, if it is a client index, the client has
+ *       to be connected.
+ *
+ * @param iIndex    Entity index
+ *
+ * @return          1 if entity is valid, 0 otherwise
+ */
 native is_valid_ent(iIndex);
 
-/* Proper origin setting, keeps updated with Half-Life engine. */
+/**
+ * Sets the origin of an entity.
+ *
+ * @note This native uses engine functions to set the origin, keeping it
```

Comment:
It uses both `SET_SIZE` and `SET_MODEL`. Either way this is just specifics, and the note should only make the user aware that there is a difference between using `EV_VEC_origin` and `entity_set_origin()`.